packages/application/src/frontend.ts

```diff
@@ -349,4 +352,17 @@ namespace Private {
    * ersatz command.
    */
   export const CONTEXT_MENU_INFO = '__internal:context-menu-info';
+
+  /**
+   * Returns whether the element is itself, or a child of, an element with the `jp-suppress-context-menu` data attribute.
+   */
+  export function suppressContextMenu(element: HTMLElement | null): boolean {
+    while (element) {
+      if (typeof element.dataset.jpSuppressContextMenu === 'string') {
+        return true;
+      }
+      element = element.parentElement;
+    }
+    return false;
```

Comment:
Maybe cleaner: `element.matches('[data-jp-suppress-context-menu] *')` ? Or possibly `element.closest('[data-jp-suppress-context-menu]') !== null`?

Reply:
Oh yeah! I didn't know these existed, thank you! the closest seems real nice. We had a [thread above](https://github.com/jupyterlab/jupyterlab/pull/7877/files/d8dd2dea420a6e132d6596ecf17c87d163de097f#diff-b2e2ad328e27adb2a492da5a97bc9436) on switching from recursive to looping.

Reply:
@saulshanabrook Just make sure whichever API you use is [supported](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest) on supported target environments. For example, `Element#closest` is not supported in IE. And further, Edge has compatibility issues.

Reply:
Edge and IE are not supported per the latest jupyterlab doc: https://jupyterlab.readthedocs.io/en/latest/getting_started/installation.html#supported-browsers

Reply:
@mlucool - thanks for pointing that out. @vidartf - thanks for pointing out `.closest` - that a great function!